arch/xtensa/src/esp32/chip_macros.h

```diff
@@ -57,8 +57,18 @@
 
 /* Definitions for the PIDs reserved for Kernel and Userspace */
 
-#define PIDCTRL_PID_KERNEL              0   /* Privileged */
-#define PIDCTRL_PID_USER                5   /* Non-privileged */
+#  define PIDCTRL_PID_KERNEL            0   /* Privileged */
+
+#ifdef CONFIG_ESP32_USER_DATA_EXTMEM
+
+/* Allocating user data in External RAM is currently limited to only using
+ * privileged PIDs (0 and 1).
```

Comment:
what's the reason of the limitation?

Reply:
does it mean the exact cause is still unknown?

i experimented pid vs psram a bit today.
when i use psram as stack, it crashed with excause 0x1c when returning from kernel.
on the other hand, load/store on psram seems working well.

btw, the TRM seems a bit unclear about how cache/mmu works in general.
is it VIVT?
does the tag include PID?

Reply:
see https://github.com/apache/nuttx/issues/13761